core/src/test/java/kafka/test/ClusterConfig.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.test;
+
+import kafka.test.annotation.Type;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+
+import java.io.File;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Represents a requested configuration of a Kafka cluster for integration testing
+ */
+public class ClusterConfig {
+
+    private final Type type;
+    private final int brokers;
+    private final int controllers;
+    private final String name;
+    private final boolean autoStart;
+
+    private final String securityProtocol;
```

Comment:
What's the advantage of making this a string rather than using the SecurityProtocol enum?

Reply:
No particular reason. We can address this when we look more into issue of multiple listeners you point out below

Reply:
If there's no reason to use a string then let's just use the `SecurityProtocol` enum.  There's no reason to create tech debt, this is all new stuff.